### PR TITLE
fix(gsoc'24): updates and bug fixes from the main simulator

### PR DESCRIPTION
**Pull request by niladrix719**
Fixes [#107](https://github.com/CircuitVerse/upptime/issues/107)

Updates and Bug fixes from the main Simulator

### CODE REVIEW

File: src/simulator/src/tutorials.js [hunk 1]
```diff
@@ -63,6 +63,15 @@ export const tour = [
             offset: 0,
         },
     },
+    {
+        element: '.testbench-manual-panel',
+        popover: {
+            title: 'Test Bench Panel',
+            description: 'This panel helps you test your circuit correctness by observing how your circuit responds under different test cases, ensuring a thorough and effective validation process.',
```
Comment (vedant-jain03) [inline src/simulator/src/tutorials.js:70]: can we make it i18n?

**Review by vedant-jain03 — CHANGES_REQUESTED** [review]

**Comment by tachyons** [comment]
I think this will undo some of the optimizations made by @JoshVarga . Why was this PR delayed ? you mentioned that all the updates from the legacy simulator are already copied, right ?

**Comment by niladrix719** [comment]
@tachyons , I guess there was some miscommunication on my end. The PR was there way back but wasn't merged before

**Comment by tachyons** [comment]
@niladrix719  Could you create separate PRs to copy changes. PR can't be reviewed in this form. Also give highest priority for this.

**Comment by niladrix719** [comment]
created Separate PRs - #345, #346, #347, #348, #349, #350, #351, #352, #353, #354, #355, #356

**Pull request closed without merging**